bin/improver-spotdb

```diff
@@ -0,0 +1,101 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Script to run spot database creation from spot data."""
+
+import iris
+import time
+import numpy as np
+
+from improver.database import SpotDatabase
+from improver.argparser import ArgParser
+
+
+def main():
+    """
+    Load in the arguments and apply the requested variant of database creation.
+    """
+    parser = ArgParser(
+        description='Create or append to a database a set of forecast cubes')
+
+    # General options:
+    parser.add_argument('-i', '--input_filepaths', nargs='*', required=True,
+                        metavar='INPUT_FILES',
+                        help='A path to an input NetCDF file to be processed.')
+    parser.add_argument('-o', '--output_filepath', required=True,
+                        metavar='OUTPUT_FILE',
+                        help='The output path for the processed database.')
+    parser.add_argument('-t', '--testing', action='store_true')
+    # Different file formats (default to SQLite DB):
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--sqlite', default=False, action='store_true',
+                       help='Create or append to a SQLite Database file.')
+    group.add_argument('--csv', default=False, action='store_true',
+                       help='The option used to create a CSV file.')
+
+    # Some options pertaining to the structure of the table.
+    structure = parser.add_argument_group(
+        'Table format options', 'Options for structure of the output table.')
+    structure.add_argument('-rows', '--row_dimension',
+                           default='time',
+                           help='A dimension to turn into the table rows')
+
+    args = parser.parse_args()
+
+    cubelist = iris.load(args.input_filepaths)
+    # TODO: Move this to the plugin itself?
+    for cube in cubelist:
+        if not cube.coords(args.row_dimension):
+            msg = ('The input cube: {} does not have a '
+                   '{} dimension.'.format(cube, columns))
+            raise CoordinateNotFoundError(msg)
+
+    if args.testing:
+        start = time.time()
+        test_constraint = iris.Constraint(**dict(realization=0))
+        cubelist = cubelist.extract(test_constraint)
+        db = SpotDatabase(cubelist, args.row_dimension)
+        db.to_dataframe()
+        print 'Time to convert {} forecasts to Dataframe:',
+        print time.time() - start
+    else:
```

Comment:
A lot of this chaining together of functions should happen in the `.process` method of the plugin.

Reply:
I can do it like this, I was just trying to separate the operations on the object for verbosity.